azure/azure.go

```diff
@@ -114,6 +241,87 @@ func (a *azureOps) InstanceID() string {
 	return a.instance
 }
 
+func (a *azureOps) InspectInstance(instanceID string) (*cloudops.InstanceInfo, error) {
+
+	instInfo := &cloudops.InstanceInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name:   a.instance,
+			ID:     a.instance,
+			Zone:   "",
+			Region: "",
+		},
+	}
+	return instInfo, nil
+}
+
+func (a *azureOps) InspectInstanceGroupForInstance(instanceID string) (*cloudops.InstanceGroupInfo, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, a.agentPoolName)
```

Comment:
Ideally, we should not be storing the managedClusterName and agentPoolName on the azureOps instance. Every request that comes to this instance of azureOps may not be for the same managedCluster or agentPool.
The API is to inspect the InstanceGroup and we should find that out here based on the given VM id.

Reply:
Inspecting Instance Group purely using VM id is not possible since cloud sdk require other info like managedCluster. If we don't want to store that info in azureOps , then we may need to change API interface definition and it's other implementations

Reply:
As discussed offline, let's leave it in the ops object for now as we have the instance stored as well. Eventually we should try to detach instance specific information from cloudops, that may require API changes.